python/audit/utils.py

```diff
@@ -0,0 +1,66 @@
+from functools import wraps
+
+from audit.models import (
+    AUDIT_OPERATION_MAP,
+    AuditLog,
+    AuditLogOperationResult,
+    AuditObject,
+)
+from cm.errors import AdcmEx
+from django.views.generic.base import View
+from rest_framework.status import is_success
+
+
+def audit(func):
+    @wraps(func)
+    def wrapped(*args, **kwargs):
+        error = None
+
+        try:
+            resp = func(*args, **kwargs)
+            status_code = resp.status_code
+        except AdcmEx as exc:
+            error = exc
+            resp = None
+            status_code = exc.status_code
```

Comment:
might be add additional logging for this situation?

Reply:
This situation can occurs when something went wrong in View code, so i believe it must be some logging there, not here. Here we should log things only about audit itself